tensorflow/compiler/tf2tensorrt/convert/weights.h

```diff
@@ -0,0 +1,252 @@
+  int64_t count() const { return count(shape_); }
+
+  // Returns the total number of elements in a weight with shape dims.
+  static int64_t count(nvinfer1::Dims dims);
+
+  size_t size_bytes() const;
+
+  string DebugString() const;
+
+  template <typename T>
+  absl::Span<const T> GetSpan() const {
+    return absl::Span<const T>(tensor_.flat<T>().data(), count());
+  }
+
+  template <typename T>
+  std::vector<T> ToVector() const {
+    auto span = GetSpan<T>();
+    return std::vector<T>(span.data(), span.data() + span.size());
+  }
+
+  nvinfer1::DataType TrtDType() const { return type_; }
+
+  // TODO(aaroey): make these private.
+  // Scalar weights are supported, a scalar constant tensor is represented via
+  // TRT_ShapedWeights::shape_ = {0, {1}}.
+  nvinfer1::Dims shape_;  // Note: shape.type[] is not used.
+
+ private:
+  // This constructor is only used by TrtWeightStore, which creates the
+  // underlying buffer.
+  TRT_ShapedWeights(nvinfer1::DataType type, nvinfer1::Dims dims,
+                    Tensor tensor);
+
+  nvinfer1::DataType type_;
+
+  // All weights should be stored inside TrtWeightStore to make sure lifetime of
+  // all the underlying tensors are available until the engine is built. For
+  // this reason, tensor_ should never be reassigned to a different value that
+  // is not already present in the TrtWeightStore.
+  Tensor tensor_;
+
+  friend class TrtWeightStore;
+};
+
+// Container for TRT_ShapedWeights. We need this container because, TRT doesn't
+// manage the lifetime of the weights buffer, it only keeps a pointer to it and
+// requires that the data referenced by the pointer be available until the
+// building of engine is complete. For more information see
+// https://docs.nvidia.com/deeplearning/sdk/tensorrt-api/c_api/classnvinfer1_1_1_weights.html
+//
+// TODO(laigd): consider adding garbage collection to the unused weights.
+class TrtWeightStore {
+ public:
+  // Get a TRT_ShapedWeights with 'type' and 'dims'.
+  TRT_ShapedWeights GetTempWeights(nvinfer1::DataType trt_type,
+                                   const nvinfer1::Dims& dims);
+
+  // Get a TRT_ShapedWeights with the same data type and dimensions as
+  // 'weights'.
+  TRT_ShapedWeights GetTempWeights(const TRT_ShapedWeights& weights) {
+    return GetTempWeights(weights.TrtDType(), weights.shape_);
+  }
+
+ private:
+  // The backend storage of the TRT_ShapedWeights.
+  std::vector<Tensor> store_;
+};
+
+// Represents a TRT-style input to a TF node, it can be either a
+// ITensorProxyPtr (representing nvinfer1::ITensor* or SimpleITensor),
+// or TRT_ShapedWeights which is compile-time constant.
+//
+// TODO(laigd): maybe rename it to TrtArgument, or mimic XlaCompiler::Argument.
+class TRT_TensorOrWeights {
+ public:
+  TRT_TensorOrWeights() {}
+  TRT_TensorOrWeights(ITensorProxyPtr);
+  TRT_TensorOrWeights(ITensorProxyPtr tensor, int batch_size);
+
+  // Constructor that makes it an ITensor, doesn't take ownership of 'tensor'.
```

Comment:
Let's replace this first line with:
//Constructs a wrapper for the given ITensor.